### DISCUSSION TITLE
Extending Shapley Values - Interaction Dependencies

### DISCUSSION THREAD

**Discussion by random-fellow** [shap/shap#2413] [Ideas]
I was wondering if you might want some help extending Shapley Values from your case of sequentially allocating value to features. To the case of allocating value to each feature based on how much value it derives from each other feature -- this would be a matrix output. I was thinking we could add some cool visualizations like a heat map for this.
https://en.wikipedia.org/wiki/Shapley_value#Value_of_a_player_to_another_player

**Comment by rlcauvin** [comment]
To visualize feature interactions, you might be interested in learned weight matrices. See the deep and cross networks (DCN) [paper](https://arxiv.org/pdf/2008.13535.pdf) and [example](https://www.tensorflow.org/recommenders/examples/dcn).

![image](https://user-images.githubusercontent.com/13562922/162331645-f82853c0-e71e-4042-8c7d-7e160e88c795.png)

I must admit, though, I haven't found a good explanation of how to interpret the matrix. The strengths of the feature crosses differ if you read left to top versus top to left, which is confusing.
